src/repr/src/row.rs

```diff
@@ -1038,19 +1055,27 @@ impl Row {
     /// byte length by calling `packer.data().len()` after pushing the desired
     /// number of datums onto the packer.
     pub unsafe fn truncate(&mut self, pos: usize) {
-        self.data.truncate(pos)
+        self.row.data.truncate(pos)
     }
 
     /// Truncates the row to contain at most the first `n` datums.
-    ///
-    /// # Panics
     pub fn truncate_datums(&mut self, n: usize) {
-        let mut iter = self.iter();
+        let mut iter = self.row.iter();
         for _ in iter.by_ref().take(n) {}
         let offset = iter.offset;
         // SAFETY: iterator offsets always lie on a datum boundary.
         unsafe { self.truncate(offset) }
     }
+
+    /// Return a copy of the current contents of the packing operation.
+    ///
+    /// The intent is that the original row's allocation can be used to pack
+    /// additional rows, to reduce the amount of interaction with the allocator.
+    pub fn freeze(&self) -> Row {
+        Row {
+            data: SmallVec::from_slice(&self.row.data),
+        }
+    }
```

Comment:
I think we want to be super clear here about the state of the `RowPacker` after this call. Maybe `freeze_but_continue` vs `freeze_and_clear`? I had expected `freeze` to clear `self.row`, but then realized that there are also good use cases where you definitely do not want that.

Reply:
I ended up deleting `freeze` in favor of calling `clone` on the underlying `Row`. It seems to work out great, and avoids introducing a new API that folks need to learn.